x-pack/plugins/apm/public/services/rest/callApi.ts

```diff
@@ -26,10 +26,38 @@ function fetchOptionsWithDebug(fetchOptions: KFetchOptions) {
   };
 }
 
+const cache = new Map();
+
+export function _clearCache() {
+  cache.clear();
+}
+
 export async function callApi<T = void>(
   fetchOptions: KFetchOptions,
   options?: KFetchKibanaOptions
 ): Promise<T> {
+  const cacheKey = JSON.stringify(fetchOptions);
+  const cacheResponse = cache.get(cacheKey);
+  if (cacheResponse) {
+    return cacheResponse;
+  }
+
   const combinedFetchOptions = fetchOptionsWithDebug(fetchOptions);
-  return await kfetch(combinedFetchOptions, options);
+  const res = await kfetch(combinedFetchOptions, options);
+
+  if (isCachable(fetchOptions)) {
+    cache.set(cacheKey, res);
+  }
+
+  return res;
+}
+
+function isCachable(fetchOptions: KFetchOptions) {
```

Comment:
I wish there was a way to make this opt-in instead of opt-out, or even just dependent on the existence of start/end -- what were the queries you were worried about omitting here if we base it on start/end?

Reply:
> what were the queries you were worried about omitting here if we base it on start/end?
Stuff like license check and index pattern. But they are loaded at bootstrap, so I don't think we need to worry about caching them actually.

I'll change the implementation, so we will ONLY cache requests that has a `start` and `end` param, and where `end` is not in the future. I agree that's better.